cloudify_starlingx/utils.py

```diff
@@ -174,6 +183,36 @@ def get_instances_of_nodes(node_id=None, node_type=None, deployment_id=None):
         raise NonRecoverableError('No node_id and no node_type provided.')
 
 
+@with_rest_client
+def get_site(site_name, rest_client):
+    try:
+        return rest_client.sites.get(site_name)
+    except CloudifyClientError:
+        return
+
+
+@with_rest_client
+def create_site(site_name, location, rest_client):
+    return rest_client.sites.create(site_name, location)
+
+
+@with_rest_client
+def update_site(site_name, location, rest_client):
+    return rest_client.sites.update(site_name, location)
+
+
+@with_rest_client
+def update_deployment_site(deployment_id, site_name, rest_client):
+    deployment = get_deployment(deployment_id)
+    if deployment.site_name == site_name:
+        return deployment
+    elif deployment.site_name:
+        return rest_client.deployments.set_site(
```

Comment:
What happens if you always call to:
rest_client.deployments.set_site(deployment_id, detach_site=True)
(even if there is no site)

Maybe we can optimize this function.